app/controllers/submission_comments_api_controller.rb

```diff
@@ -18,12 +18,84 @@
+# This API can be used to edit and delete submission comments.
+# It can also be used to create files to attach to submission comments. The
 # submission comments themselves can be created using the Submissions API.
 class SubmissionCommentsApiController < ApplicationController
   before_action :require_context
 
   include Api::V1::Attachment
+  include Api::V1::SubmissionComment
+
+  # @API Edit a submission comment
+  #
+  # Edit the given submission comment.
+  #
+  # @argument comment [String]
+  #   If this argument is present, edit thet text of a comment.
+  #
+  # @argument attachment_ids[] [Integer]
+  #   Attach files to this comment that were previously uploaded using the
+  #   Submission Comment API's files action.
+  #
+  # @argument media_comment_id [String]
+  #   Add an audio/video comment to the submission. Media comments can be added
+  #   via this API, however, note that there is not yet an API to generate or
+  #   list existing media comments, so this functionality is currently of
+  #   limited use.
+  #
+  # @argument media_comment_type [String, "audio"|"video"]
+  #   The type of media comment being added.
+  #
+  # @returns SubmissionComment
+  def update
+    submission_comment = SubmissionComment.find(params[:id])
+    if authorized_action(submission_comment, @current_user, :update)
+      submission_comment.updating_user = @current_user
+      submission_comment.reload unless submission_comment.update(submission_comment_params)
+
+      return render json: anonymous_moderated_submission_comments_json(
+        assignment: submission_comment.submission.assignment,
+        course: @context,
+        current_user: @current_user,
+        avatars: service_enabled?(:avatars),
+        submission_comments: [submission_comment],
+        submissions: [submission_comment.submission]
+      )[0]
+    end
+  end
+
+  def submission_comment_params
+    params.permit(:comment, :attachment_ids, :media_comment_id, :media_comment_type)
+  end
+
+  # @API Delete a submission comment
+  #
+  # Delete the given submission comment.
+  #
+  # @example_request
+  #     curl https://<canvas>/api/v1/courses/<course_id>/assignments/<assignment_id>/submissions/<user_id>/comments/<id> \
+  #          -X DELETE \
+  #          -H 'Authorization: Bearer <token>'
+  # @returns SubmissionComment
+  def destroy
+    submission_comment = SubmissionComment.find(params[:id])
+    if authorized_action(submission_comment, @current_user, :delete)
+      comment_data = anonymous_moderated_submission_comments_json(
+        assignment: submission_comment.submission.assignment,
+        course: @context,
+        current_user: @current_user,
+        avatars: service_enabled?(:avatars),
+        submission_comments: [submission_comment],
+        submissions: [submission_comment.submission]
+      )[0]
+
+      submission_comment.updating_user = @current_user
+      submission_comment.destroy
```

Comment:
We'd prefer if you go with `.destroy!` here instead of `.destroy`. The former will raise an error if there are issues destroying the record.